src/parquet/schema.h

```diff
@@ -261,16 +267,23 @@ class PARQUET_EXPORT GroupNode : public Node {
       const NodeVector& fields, LogicalType::type logical_type = LogicalType::NONE,
       int id = -1)
       : Node(Node::GROUP, name, repetition, logical_type, id), fields_(fields) {
+    field_name_to_idx_.clear();
+    auto field_idx = 0;
     for (NodePtr& field : fields_) {
       field->SetParent(this);
+      field_name_to_idx_[field->name()] = field_idx++;
     }
   }
 
   NodeVector fields_;
   bool EqualsInternal(const GroupNode* other) const;
 
+  // Mapping between field name to the field index
+  std::unordered_map<std::string, int> field_name_to_idx_;
```

Comment:
I think its ok. I dont think its a big overhead.
I have been tempted to use the same map on `SchemaDescriptor` used for mapping paths to leaves for all nodes and use it instead but this way seemed easier to navigate the tree, even where you don't have the descriptor object.